include/libdnf/base/goal.hpp

```diff
@@ -249,7 +249,7 @@ class Goal {
     void add_rpm_reason_change(
         const std::string & spec,
         const libdnf::transaction::TransactionItemReason reason,
-        const std::optional<std::string> & group_id,
+        const std::string & group_id = {},
```

Comment:
Shall we have another method for the reason group ?

Reply:
I suppose we can also use the overload approach here, but in this case, is there anything wrong about having the default empty `group_id`? Is there some usecase that we want to treat empty `group_id` values coming through this call differently?

Reply:
Empty string shouldn't be any problem. It is only valid and used if the new reason is "GROUP".